MHVLib/Connection/MHVConnection.m

```diff
@@ -255,13 +266,19 @@ - (void)executeMethodRequest:(MHVHttpServiceRequest *)request
         }
         else
         {
+            if (request.serviceOperation.cache)
+            {
+                [request.serviceOperation.cache setObject:response forKey:cacheKey];
+            }
```

Comment:
I think this is done in the wrong place - I believe you cannot save the response until after it is parsed and we determine there are no errors. For XML responses, everything is a 200 SUCCESS, and there is a status code in the payload that may or may not have an error.